driver/controller.go

```diff
@@ -441,21 +441,34 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	if req.GetSourceVolumeId() == "" {
 		return nil, status.Error(codes.InvalidArgument, "snapshot source volume ID must be provided")
 	}
+
 	log := logWithServerContext(ctx, d.log)
 	log.Info("getting storage backup by name")
+
 	s, err := d.svc.getStorageBackupByName(ctx, req.GetName())
 	if err != nil && !errors.Is(err, errUpCloudStorageNotFound) {
-		return nil, status.Errorf(codes.Internal, "createsnapshot failed with: %s", err.Error())
+		return nil, status.Errorf(codes.Internal, "CreateSnapshot failed with: %s", err.Error())
 	}
 
 	if s != nil && s.Origin != req.GetSourceVolumeId() {
 		return nil, status.Error(codes.AlreadyExists, "snapshot already exists with different source volume ID")
 	}
+
 	if s == nil {
-		log.Info("creating strorage backup")
+		// check that a backup creation is not currently in progress
+		backupInProgress, err := d.svc.checkIfBackingUp(ctx, req.GetSourceVolumeId())
```

Comment:
Not a big one, but maybe we could keep interface smaller and do this logic inside `createStorageBackup` and return custom error if backup is already running? :thinking:

Reply:
Good suggestion! I just pushed some updates to slim the interface down.